File: src/main/java/com/example/treemenu/controller/MenuController.java
package com.example.treemenu.controller;

import com.example.treemenu.service.MenuService;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;

@Controller
public class MenuController {
    private MenuService service;

    public MenuController(MenuService service) {
        this.service = service;
    }

    @GetMapping
    public String getMenu(Model model) {
        model.addAttribute("rootMenuList", service.getMenuTree());
        return "index";
    }
}

File: src/main/kotlin/com/example/treemenu/controller/MenuController.kt
package com.example.treemenu.controller

import com.example.treemenu.service.MenuService
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping

@Controller
class MenuController(private val service: MenuService) {
    @GetMapping
    fun getMenu(model: Model): String {
        model.addAttribute("rootMenuList", service.menuTree)
        return "index"
    }
}